### ISSUE TITLE
准确率问题

### ISSUE THREAD

**Issue by HeTaoya** [lylylylylyly/PRM-FSRE#1]
大佬你好，首先感谢你的开源项目。但是在运行你的代码时我遇到了一个问题，我始终无法获得一个较高的准确率，我使用的是bert-base-uncased作为编码器，proto作为训练的模型，准确率只有65%。

**Comment by lyyang01** [comment]
你好，我把val_step改成200做了下测试，我这边在val dataset上的准确率应该是正常的，我把models/__init__.py做了一下更新。可以检查一下有没有使用到proto_7_2这个文件，再跑多一些step步之后再看准确率
![1698137715424](https://github.com/lylylylylyly/PRM-FSRE/assets/8745138/4bcc0c9a-aef3-4e0a-ac26-56f8fe5f088d)

**Comment by HeTaoya** [comment]
大佬您好：
准确率还是只有那么多，
我做了一点点代码上的修改（framwork.py），我的显存有点不够用（3090），训练这个模型爆显存。
分别是在271行模型训练前加了不计算梯度的命令：with torch.no_grad()，和310损失反向传播前加了：loss.requires_grad_(Ture)
这会导致严重的后果吗？
谢谢大佬百忙之中解答我这个小趴菜的问题（十分谢谢）。

**Comment by lyyang01** [comment]
是的，在训练前加 with torch.no_grad()是有影响的，相当于模型没有进行训练，所以准确率一直上不去。我不太清楚loss.requires_grad(True)是否会抵消with torch.no_grad()的影响，但从结果来看好像是没有。如果显存不够也许可以考虑使用apex包，采用半精度训练模型

**Comment by HeTaoya** [comment]
你好,我在你的项目上进行了一些改动,但是好像出现了一些问题.具体情况是这样的:
例如:10-N-1-w.
经过长时间训练后,训练集,验证集,测试集的准确率都非常的高:分别是99.5%左右,99%左右,98.5%左右.
而如果在训练2000step后就停止,训练集,验证集的准确率基本上没有变化,测试集准确率维持在96.3%左右.
其他的5-1 5-5 10-10的情况基本上一致的,完整训练后,测试集准确率都非常的高(基本上维持在98%左右,实验了多次)
这是一种什么原因呢?如果想评估模型性能的话,应该怎么办呢?希望大佬能够解惑.






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "lylylylylyly/PRM-FSRE"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年10月25日(星期三) 上午9:40
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [lylylylylyly/PRM-FSRE] 准确率问题 (Issue #1)





 
是的，在训练前加 with torch.no_grad()是有影响的，相当于模型没有进行训练，所以准确率一直上不去。我不太清楚loss.requires_grad(True)是否会抵消with torch.no_grad()的影响，但从结果来看好像是没有。如果显存不够也许可以考虑使用apex包，采用半精度训练模型

**Comment by lyyang01** [comment]
这可能需要知道你代码改动的细节，从结果上来看比较奇怪。测试集我记得是没有开源标签的，你需要运行run_submit.sh，然后将得到的文件提交到这个竞赛上面：https://competitions.codalab.org/competitions/27980
